packages/@aws-cdk/aws-ivs-alpha/lib/recording-configuration.ts

```diff
@@ -0,0 +1,210 @@
+import { CfnRecordingConfiguration } from 'aws-cdk-lib/aws-ivs';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { Duration, Fn, IResource, Resource, Stack, Token } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import { RenditionConfiguration } from './rendition-configuration';
+import { ThumbnailConfiguration } from './thumbnail-configuration';
+
+/**
+ * Properties of the IVS Recording configuration
+ */
+export interface RecordingConfigurationProps {
+  /**
+   * S3 bucket where recorded videos will be stored.
+   */
+  readonly bucket: IBucket;
+
+  /**
+   * The name of the Recording configuration.
+   * The value does not need to be unique.
+   *
+   * @default - auto generate
+   */
+  readonly recordingConfigurationName?: string;
+
+  /**
+   * If a broadcast disconnects and then reconnects within the specified interval,
+   * the multiple streams will be considered a single broadcast and merged together.
+   *
+   * `recordingReconnectWindow` must be between 0 and 300 seconds
+   *
+   * @default - 0 seconds (means disabled)
+   */
+  readonly recordingReconnectWindow?: Duration;
+
+  /**
+   * A rendition configuration describes which renditions should be recorded for a stream.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ivs-recordingconfiguration-renditionconfiguration.html
+   *
+   * @default - no rendition configuration
+   */
+  readonly renditionConfiguration?: RenditionConfiguration;
+
+  /**
+   * A thumbnail configuration enables/disables the recording of thumbnails for a live session and controls the interval at which thumbnails are generated for the live session.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ivs-recordingconfiguration-thumbnailconfiguration.html
+   *
+   * @default - no thumbnail configuration
+   */
+  readonly thumbnailConfiguration?:ThumbnailConfiguration;
+}
```

Comment:
Thanks for making the requested changes @mazyu36! I notice now though that the properties that are contained within the `renditionConfiguration` and `thumbnailConfiguration` no longer have defaults because the new default is that the configuration is not provided. Is this intentional or should we still set a default for these properties somewhere?

Reply:
@paulhcsun 
Thanks.
The lack of default values is intentional. Is it correct to understand that you're suggesting we should document somewhere what the default values are when @param is omitted? If so, I plan to add this information to the @param description.

Reply:
I added an explanation regarding the default value. Please let me know if it doesn’t match your intention.

Reply:
Yes the documentation is appreciated! But also I wanted to check if this default is something that gets set by the service/cloudformation fills it in, since we're not explicitly setting it as the default anywhere.

Reply:
Thanks. 
Those default values are provided by CFn.
So I've not explicitly set.

Reply:
Gotcha, I'm good to approve then. Thank you for working on this @mazyu36!!